### DISCUSSION TITLE
How to write the following code in webpack4

### DISCUSSION THREAD

**Discussion by enkialtria** [webpack/webpack#16817] [Q&A]
chainWebpack: config => {
        config.merge({
          resolve: {
            conditionNames: ['require', 'node']
          }
        })
}

**Comment by alexander-akait** [comment]
Please provide more information to help you

**Comment by enkialtria** [comment]
> 
![F3E67CC4-4723-4C04-8827-031FF82C49E8](https://user-images.githubusercontent.com/45117405/224801917-a1619c50-ba24-4a8b-a161-de44634f57f1.png)

**Reply by alexander-akait** [reply]
This option is not supported by webpack v4, sorry, you need to update webpack to v5

**Reply by enkialtria** [reply]
I try write:  

resolve: {
    // ...
    conditionNames: ['require', 'node']
}


But  error
![358534E6-290E-43E2-B123-28F68DA9ED30](https://user-images.githubusercontent.com/45117405/224803978-1b4f495e-587c-4bf7-8de0-18a883996142.png)
![F8CEAA25-97D4-4847-AD5D-C43F39D52398](https://user-images.githubusercontent.com/45117405/224803987-3d411c01-33e5-4404-a6f3-e5013b43ef25.png)

**Reply by enkialtria** [reply]
i  use webpack5 to build a web url, and use iframe.
It is a editor,Form https://github.com/baidu/amis/issues/5274

**Reply by enkialtria** [reply]
thank you

**Reply by alexander-akait** [reply]
Please write it in `chainWebpack` section, i.e.
```js
config.resolve.conditionName = ['require', 'node'];
```

**Reply by alexander-akait** [reply]
You can move it in `configureWebpack` section in `resolve` section

**Reply by enkialtria** [reply]
Ok,I'll try write it in this js
https://github.com/yangzongzhuan/RuoYi-Vue/blob/master/ruoyi-ui/vue.config.js
But tomorrow
thank you again

**Reply by enkialtria** [reply]
no success in webpack4

**Comment by enkialtria** [comment]
![1678799741843](https://user-images.githubusercontent.com/45117405/225012598-78ec9fcf-5b18-4310-abaf-817a4b4302e2.png)
or
![1678799771022](https://user-images.githubusercontent.com/45117405/225012632-4119b4a7-7799-4592-a046-485d4baf9af3.png)
conditionName  or conditionNames

**Reply by enkialtria** [reply]
no success

**Reply by enkialtria** [reply]
I can only use webpack5

**Reply by alexander-akait** [reply]
> no success

What do you mean?

**Reply by enkialtria** [reply]
I write
config.resolve.conditionName = ['require', 'node']
in webpack4
Didn't solve the problem

**Reply by enkialtria** [reply]
so i think i can only use webpack5

**Reply by alexander-akait** [reply]
`conditionName` doesn't work for webpack v4 as I said above

**Reply by enkialtria** [reply]
Ok,thank you
